src/machine/machine_k210.go

```diff
@@ -0,0 +1,166 @@
+// +build k210
+
+package machine
+
+import (
+	"device/kendryte"
+	"runtime/interrupt"
+)
+
+func CPUFrequency() uint32 {
+	return 390000000
+}
+
+type PinMode uint8
+
+const (
+	PinInput PinMode = iota
+	PinInputPullUp
+	PinInputPullDown
+	PinOutput
+)
+
+type fpioaPullMode uint8
+
+const (
+	fpioaPullNone fpioaPullMode = iota
+	fpioaPullDown
+	fpioaPullUp
+)
+
+func (p Pin) fpioaSetIOPull(pull fpioaPullMode) {
+	switch pull {
+	case fpioaPullNone:
+		kendryte.FPIOA.IO[uint8(p)].ClearBits(kendryte.FPIOA_IO_PU & kendryte.FPIOA_IO_PD)
+	case fpioaPullUp:
+		kendryte.FPIOA.IO[uint8(p)].SetBits(kendryte.FPIOA_IO_PU)
+		kendryte.FPIOA.IO[uint8(p)].ClearBits(kendryte.FPIOA_IO_PD)
+	case fpioaPullDown:
+		kendryte.FPIOA.IO[uint8(p)].ClearBits(kendryte.FPIOA_IO_PU)
+		kendryte.FPIOA.IO[uint8(p)].SetBits(kendryte.FPIOA_IO_PD)
+	}
+}
+
+// Configure this pin with the given configuration.
+func (p Pin) Configure(config PinConfig) {
+	var input bool
+
+	kendryte.FPIOA.IO[uint8(p)].SetBits(kendryte.FPIOA_IO_OE_EN | kendryte.FPIOA_IO_IE_EN | kendryte.FPIOA_IO_ST | kendryte.FPIOA_IO_DS_Msk)
+	switch config.Mode {
+	case PinInput:
+		p.fpioaSetIOPull(fpioaPullNone)
+		input = true
+
+	case PinInputPullUp:
+		p.fpioaSetIOPull(fpioaPullUp)
+		input = true
+
+	case PinInputPullDown:
+		p.fpioaSetIOPull(fpioaPullDown)
+		input = true
+
+	case PinOutput:
+		p.fpioaSetIOPull(fpioaPullDown)
+		input = false
+
+	}
+
+	if p >= D0 && p <= D7 {
```

Comment:
```suggestion
	if p >= P08 && p <= P15 {
```

These `D0` etc pins are board specific.

Reply:
I changed this to use the chip pins. thank you